tests/Valkey.Glide.IntegrationTests/ReadFromEndToEndIntegrationTests.cs

```diff
@@ -0,0 +1,992 @@
+        Assert.NotNull(connectionMultiplexer);
+
+        // Parse the original configuration to verify ReadFrom was set correctly
+        var parsedConfig = ConfigurationOptions.Parse(connectionString);
+        Assert.NotNull(parsedConfig.ReadFrom);
+        Assert.Equal(expectedStrategy, parsedConfig.ReadFrom.Value.Strategy);
+        Assert.Equal(expectedAz, parsedConfig.ReadFrom.Value.Az);
+
+        // Verify the configuration reaches the underlying client by testing functionality
+        var database = connectionMultiplexer.GetDatabase();
+        Assert.NotNull(database);
+
+        // Test a basic operation to ensure the connection works with ReadFrom configuration
+        await database.PingAsync();
+
+        // Test data operations to verify the ReadFrom configuration is active
+        string testKey = Guid.NewGuid().ToString();
+        string testValue = "cluster-end-to-end-test";
+        await database.StringSetAsync(testKey, testValue);
+        string? retrievedValue = await database.StringGetAsync(testKey);
+        Assert.Equal(testValue, retrievedValue);
+
+        // Cleanup
+        await database.KeyDeleteAsync(testKey);
+        connectionMultiplexer.Dispose();
+    }
+
+    [Fact]
+    public async Task EndToEnd_ConnectionStringWithoutReadFrom_DefaultsToNullInFFILayer()
+    {
+        // Arrange
+        string connectionString = $"{TestConfiguration.STANDALONE_HOSTS[0].host}:{TestConfiguration.STANDALONE_HOSTS[0].port},ssl={TestConfiguration.TLS}";
+
+        // Act
+        var connectionMultiplexer = await ConnectionMultiplexer.ConnectAsync(connectionString);
+
+        // Assert - Verify connection was created successfully
+        Assert.NotNull(connectionMultiplexer);
+
+        // Parse the original configuration to verify ReadFrom is null (default behavior)
+        var parsedConfig = ConfigurationOptions.Parse(connectionString);
+        Assert.Null(parsedConfig.ReadFrom);
+
+        // Test a basic operation to ensure the connection works without ReadFrom configuration
+        var database = connectionMultiplexer.GetDatabase();
+        await database.PingAsync();
+
+        // Test data operations to verify default behavior works
+        string testKey = Guid.NewGuid().ToString();
+        string testValue = "default-behavior-test";
+        await database.StringSetAsync(testKey, testValue);
+        string? retrievedValue = await database.StringGetAsync(testKey);
+        Assert.Equal(testValue, retrievedValue);
+
+        // Cleanup
+        await database.KeyDeleteAsync(testKey);
+        connectionMultiplexer.Dispose();
+    }
+
+    [Theory]
+    [InlineData("primary")]
+    [InlineData("PREFERREPLICA")]
+    [InlineData("azaffinity")]
+    [InlineData("AzAffinityReplicasAndPrimary")]
+    public async Task EndToEnd_ConnectionString_CaseInsensitiveReadFromParsing(string strategyString)
+    {
+        // Arrange
+        var expectedStrategy = strategyString.ToLowerInvariant() switch
```

Comment:
I already seen this switch somewhere - can you re-use it?

Reply:
Swapped with an Enum.Parse<T>